gateway/hydrator.go

```diff
@@ -76,25 +81,73 @@ type EndpointHydrator struct {
 // to start generating and sending endpoint check requests.
 func (eph *EndpointHydrator) Start() error {
 	if eph.Protocol == nil {
-		return errors.New("an instance of Protocol must be provided.")
+		return errors.New("an instance of Protocol must be provided")
 	}
 
 	if len(eph.ActiveQoSServices) == 0 {
 		return errors.New("at least one QoS instance must be provided to the endpoint hydrator to start sending check requests")
 	}
 
 	go func() {
+		// Wait for the protocol to be healthy before starting hydrator
+		eph.waitForProtocolHealth()
+
+		// Bootstrap QoS data with initial checks before starting regular interval
+		eph.bootstrapInitialQoSData() // Run 5 initial checks
+
+		// Start regular interval checks
 		ticker := time.NewTicker(eph.RunInterval)
 		for {
-			eph.run()
+			// run the hydrator and get the successful service checks
+			successfulServiceChecks := eph.run()
+
+			// update the hydrator's health status
+			eph.healthStatusMutex.Lock()
+			eph.isHealthy = eph.getHealthStatus(successfulServiceChecks)
+			eph.healthStatusMutex.Unlock()
+
+			// wait for the next interval
 			<-ticker.C
 		}
 	}()
 
 	return nil
 }
 
-func (eph *EndpointHydrator) run() {
+// waitForProtocolHealth blocks until the Protocol reports as healthy.
+// This ensures that the hydrator only starts running once the underlying
+// protocol layer is ready.
+func (eph *EndpointHydrator) waitForProtocolHealth() {
+	eph.Logger.Info().Msg("waitForProtocolHealth: waiting for protocol to become healthy before starting hydrator")
+
+	for !eph.Protocol.IsAlive() {
+		eph.Logger.Info().Msg("waitForProtocolHealth: protocol not yet healthy, waiting...")
+		time.Sleep(1 * time.Second)
+	}
+
+	eph.Logger.Info().Msg("waitForProtocolHealth: protocol is now healthy, hydrator can proceed")
+}
+
+// bootstrapInitialQoSData runs a specified number of hydrator checks immediately
+// to quickly bootstrap QoS data on startup. This helps to identify and filter out
+// invalid endpoints as soon as possible. The hydrator's health status is only set
```

Comment:
I do agree that HYdrator's health status determination logic needs an update, but not quite sure this specific change improves it.